analyses/collapse-rnaseq/cwl/scripts/liftover_collapse_rnaseq.py

```diff
@@ -0,0 +1,180 @@
+"""
+Script to use a GTF and expression matrix with ENSG to liftover gene symbols
+"""
+
+import argparse
+import gzip
+import re
+import sys
+from statistics import mean
+
+
+def update_gtf_dict(in_dict, in_text):
+    """
+    Update ensg - sym ID dictionary on gene entries only
+    """
+    data = in_text.strip('\n').split('\t')
+    if data[tx_type] == 'gene':
+        try:
+            gene_id = id_find.match(data[tx_desc]).group(1)
+            gene_name = name_find.match(data[tx_desc]).group(1)
+            in_dict[gene_id] = gene_name
+        except Exception as e:
+            sys.stderr.write(str(e) + '\n')
+            sys.stderr.write("Failed to parse gene ID and name from " + in_text)
+
+
+def liftover(in_data, out_lift, n, l):
+    """
+    Update gene symbol if found in dict, leave alone if not
+    """
+    if l % n == 0:
+        sys.stderr.write("Processed " + str(l) + " lines\n")
+        sys.stderr.flush()
+    data = in_data.rstrip('\n').split('\t')
+    cur_id = data[g_idx].split('.')[0]
+    if cur_id in gtf_dict:
+        data[n_idx] = gtf_dict[cur_id]
+    gene_sym = data[n_idx]
+    if gene_sym not in sym_ct:
+        sym_ct[gene_sym] = 1
+    else:
+        sym_ct[gene_sym] += 1
```

Comment:
You're building this dict for seemingly no reason. It's not used anywhere else in this function and you do not return it.

Reply:
Loosey goosey python global variable access obscures this. Will add to def for clarity, but it's purpose is defined here: 
```python
    # track sym occurrences if collapse flag given for later processing
    sym_ct = {}
```